### PR TITLE
new apple frameworks test #6499

### PR DESCRIPTION
**Pull request by SSE4**
Changelog: Bugfix: improve Apple frameworks lookups with CMake integration
Docs: omit

updated version of #6499
closes: #6416
code from https://github.com/Paultergeist/conan_apple_framework

- [ ] Refer to the issue that supports this Pull Request.
- [ ] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [ ] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [ ] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

**Comment by SSE4** [comment]
@memsharded I got further on this, it now builds for me, however, it fails to run:
```
dyld: Library not loaded: @rpath/hello.framework/Versions/A/hello
  Referenced from: /private/var/folders/vq/v4_w2g4j5k1170y70pgfktzr0000gn/T/tmpns2pz_0mconans/path with spaces/test_package/build/f4c7a70248efcdb8d65e75e42ba42fadfc285e1b/bin/timer
  Reason: image not found
ERROR: mylibrary/1.0 (test package): Error in test() method, line 14
	self.run("bin/timer", run_environment=True)
	ConanException: Error -6 while executing DYLD_LIBRARY_PATH="" DYLD_FRAMEWORK_PATH="" bin/timer
```
this is very suspicious, as testing with real project (SFML) passes.
right now I fail to understand why run_environment doesn't see **framework_paths**?
it turns our that dumping such values results in incorrect things:
```
    def test(self):
                    self.output.info("%s" % self.deps_cpp_info["mylibrary"].frameworks)
                    self.output.info("%s" % self.deps_cpp_info["mylibrary"].framework_paths)
                    self.output.info("%s" % self.deps_cpp_info["mylibrary"].frameworkdirs)
```

```
mylibrary/1.0 (test package): Running test()
mylibrary/1.0 (test package): ['hello']
mylibrary/1.0 (test package): []
mylibrary/1.0 (test package): ['Frameworks']
```
somehow, **framework_paths**/**frameworkdirs** are lost in RunEnvironment generator... but values were passed correctly to the CMake generator, that puzzles me a lot.

**Comment by SSE4** [comment]
okay, it turns out problem was in txt generator (surprisingly). it misused **framework_paths** vs **framework_dirs**.

File: conans/client/generators/cmake_common.py [hunk 1]
```diff
@@ -747,29 +747,30 @@ class CMakeCommonMacros:
...
                     endif()
                 endif()
                 foreach(_FRAMEWORK ${FRAMEWORKS})
                     # https://cmake.org/pipermail/cmake-developers/2017-August/030199.html
-                    find_library(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND NAME ${_FRAMEWORK} PATHS ${CONAN_FRAMEWORK_DIRS})
+                    find_library(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND NAME ${_FRAMEWORK} PATHS ${CONAN_FRAMEWORK_DIRS${SUFFIX}})
                     if(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND)
                         list(APPEND ${FRAMEWORKS_FOUND} ${CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND})
                     else()
+                        message(FATAL_ERROR "Framework library ${_FRAMEWORK} not found in paths: ${CONAN_FRAMEWORK_DIRS${SUFFIX}}")
```
Comment (memsharded) [inline conans/client/generators/cmake_common.py]: Repeated FATAL_ERROR? Second one never executes?

Comment (SSE4) [reply conans/client/generators/cmake_common.py]: good catch, I'll remove the second one

**Comment by SSE4** [comment]
the second test has been added.

**Comment by SSE4** [comment]
NOTE: this is not a regression, it wasn't working as expected from the inception.

**Comment by memsharded** [comment]
Many thanks!

**Comment by jacobjiangwei** [comment]
hey there, I have tried latest build Conan version 1.31.1
and same issue still occurs.

I made a swift dynamic framework for our own testing purpose and installed to the test project.

The test project can import the framework and compile successfuly, but run instantly into crashes:

dyld: Library not loaded: @rpath/swiftTestFramework.framework/swiftTestFramework
  Referenced from: /private/var/containers/Bundle/Application/7589D3D1-8813-4195-8FD6-6CA041AB6439/test.app/test
  Reason: image not found
dyld: launch, loading dependent libraries
DYLD_LIBRARY_PATH=/usr/lib/system/introspection
DYLD_INSERT_LIBRARIES=/Developer/usr/lib/libBacktraceRecording.dylib:/Developer/usr/lib/libMainThreadChecker.dylib:/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib


Can you point me the right place to this issue? Thanks

**Pull request merged**